### DISCUSSION TITLE
create own logdriver

### DISCUSSION THREAD

**Discussion by tiolan** [containers/podman#14535] [Q&A] [answered]
Hey there,

I would like to create an own, custom logdriver for podman.
Is this officially supported by some kind of plug-in system / library or would it require to change the source code of podman itself?

Thanks

**Comment by mheon** [comment, marked as answer]
There is presently no plugin system - all log drivers must be implemented in Conmon. However, we are investigating a new version of Conmon that has been rewritten in Rust, and a log driver plugin system there would be a neat feature.

**Reply by z0rb** [reply]
@mheon This this lead anywhere?

There are multiple requests for similar things, e.g.: 

https://github.com/containers/conmon/issues/552
https://github.com/containers/conmon/issues/340

**Reply by mheon** [reply]
conmon-rs exists but adopting is has never managed to make its way high enough up the priority stack to actually be done, and it doesn't support a plugin system for log drivers at the moment either (though it'd be a lot easier to do there than in the current, C conmon). Optimistically we'll get conmon-rs integrated at some point in 2025 as a non-default option and then transition to it as default with Podman 6 in May of '26, but that is assuming nothing else important comes up that requires us to switch focus.
